libdeflate: register static build as test (ref: #144207)

KAction opened the pull request:
###### Motivation for this change

Fix static build of libdeflate

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Commit:
libdeflate: register static build as test (ref: #144207)

KAction commented:
I know that static build is broken and I did not fix it yet. I want to make sure that with static build as test CI will fail.

SuperSandro2000 commented on pkgs/development/libraries/libdeflate/default.nix:
```diff
@@ -30,4 +30,5 @@ stdenv.mkDerivation rec {
     platforms = platforms.unix;
     maintainers = with maintainers; [ orivej ];
   };
+  passthru.tests.static = pkgsStatic.libdeflate;
```
passthru should be before meta.

KAction replied:
Ok, I'll fix.

KAction commented:
Any idea why `libdeflate, libdeflate.passthru.tests on x86_64-linux — Failure` is marked as "neutral", not as failure?

Commit:
libdeflate: fix static build

Patch build system to make building of shared libraries optional, and
disable it when host system is static.

Commit:
libdeflate: add myself as maintainer of the package

I am not really interested in maintaining package pe-se, but since static
version of it is a dependency of something I care about, I want to subscribe
to all changes that can potentially break it.

KAction commented:
I also added myself as maintainer to get notifications about changes. Do we have concept of "subscribe to particular path in repository"?

SuperSandro2000 commented:
> I also added myself as maintainer to get notifications about changes. Do we have concept of "subscribe to particular path in repository"?

Only GitHub Codeowners but I think you need commit permissions for that to work.

SuperSandro2000 commented on pkgs/development/libraries/libdeflate/default.nix:
```diff
@@ -1,6 +1,19 @@
+  staticArgs = {
+    patches = [
+      (fetchpatch {
+        url = "https://github.com/ebiggers/libdeflate/pull/137.diff";
```
```suggestion
        url = "https://github.com/ebiggers/libdeflate/commit/ee4d18872bfe09a32cfd031c716b9069a04a50a0.patch";
```

SuperSandro2000 commented on pkgs/development/libraries/libdeflate/default.nix:
```diff
@@ -23,11 +36,12 @@ stdenv.mkDerivation rec {
+    maintainers = with maintainers; [ orivej kaction ];
   };
-}
+} // (if stdenv.hostPlatform.isStatic then staticArgs else { }))
```
Can we just throw this onto staging to avoid this hackery?

KAction replied:
Done.

KAction commented:
What is the process to get commit bit?

mkg20001 commented:
If you're active enough you can apply through this issue https://github.com/NixOS/nixpkgs/issues/50105

SuperSandro2000 commented:
> If you're active enough you can apply through this issue 50105

Active is not a real hard requirement. I think the only requirements are that you are a trusted member of the community for a reasonable amount of time and you should use the commit bit somewhat regularly and don't be to inactive.
